src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1070,6 +1075,22 @@ private UsernamePasswordAuthenticationToken loginAndSetUserData(
             user.setFullName(fullName);
         }
 
+        // Set avatar if possible
+        try {
+            String avatarUrl = determineStringField(avatarFieldExpr, idToken, userInfo);
+            if (avatarUrl != null) {
+                LOGGER.finest("Avatar url is: " + avatarUrl);
+                OicAvatarProperty.AvatarImage avatarImage = new OicAvatarProperty.AvatarImage(avatarUrl);
+                OicAvatarProperty oicAvatarProperty = new OicAvatarProperty(avatarImage);
+                user.addProperty(oicAvatarProperty);
+            } else {
+                LOGGER.finest("No avatar URL found");
+            }
+
+        } catch (Exception e) {
```

Comment:
is it still valid to have this in a try/catch now we are not downloading, / decrypting the picture?